### PR TITLE
dhat memory profiling

### PR DESCRIPTION
**Pull request by jonas-schievink**
Unfortunately this does not currently work due to https://github.com/nnethercote/dhat-rs/issues/4

### CODE REVIEW

**Comment by kjeremy** [comment]
https://github.com/rust-analyzer/rust-analyzer/issues/1252#issuecomment-741912540

File: Cargo.lock [hunk 1]
```diff
@@ -480,6 +494,15 @@ version = "0.3.3"
...
 checksum = "3dcaa9ae7725d12cdb85b3ad99a434db70b468c09ded17e012d86b5c1010f7a7"
 
+[[package]]
+name = "fxhash"
+version = "0.2.1"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "c31b6d751ae2c7f11320402d34e41349dd1016f8d5d45e48c4312bc8625af50c"
+dependencies = [
+ "byteorder",
+]
```
Comment (matklad) [inline Cargo.lock:504]: I wonder if dhat can dedup that with rustc_hash...

Comment (matklad) [reply Cargo.lock:504]: https://github.com/nnethercote/dhat-rs/pull/8

**Comment by jonas-schievink** [comment]
I've let `analysis-stats` run for several hours now, and it is now consuming 30 GB of RAM with no end in sight. Looks like valgrind DHAT is still a lot more efficient, so closing for now.

**Comment by matklad** [comment]
cc @nnethercote, this might be a fun stress test for dhat

**Pull request closed without merging**